shoop/admin/modules/orders/views/shipment.py

```diff
@@ -17,15 +19,68 @@
 from shoop.admin.toolbar import PostActionButton, Toolbar
 from shoop.admin.utils.forms import add_form_errors_as_messages
 from shoop.admin.utils.urls import get_model_url
+from shoop.apps.provides import get_provide_objects
 from shoop.core.excs import NoProductsToShipException
-from shoop.core.models import Order, Product, Supplier
+from shoop.core.models import Order, Product, Shipment, Supplier
+
+FORM_MODIFIER_PROVIDER_KEY = "admin_extend_create_shipment_form"
+
+
+class ShipmentFormModifier(six.with_metaclass(abc.ABCMeta)):
+    def get_extra_fields(self, order):
+        """
+        Extra fields for shipment creation form
+
+        :param order: Order linked to form
+        :type order: shoop.core.models.Order
+        :return: List of extra fields that should be added to form.
+        Tuple should contain field name and Django form field.
+        :rtype: list[tuple(str,django.forms.Field)]
+        """
+        pass
+
+    def clean_hook(self, cleaned_data):
+        """
+        Extra clean for shipment creation form
+
+        :param cleaned_data: form cleaned data
+        :type cleaned_data: dict
+        :return: List of errors that should be handled.
+        Tuple should contain field name and error message.
+        :rtype: list[tuple(str,str)]
+        """
+        pass
+
+    def form_valid_hook(self, form, shipment):
+        """
+        Extra form valid handler for shipment creation form
```

Comment:
Now that I look more carefully, I see that this is actually called from the view, not from the form. Therefore the last word in this title is incorrect. It should be "shipment creation view.". You're also missing the period: docstring title should end with period.